src/pypromice/process/write.py

```diff
@@ -148,46 +168,42 @@ def writeNC(outfile, Lx, col_names=None):
         names = [c for c in col_names if c in list(Lx.keys())]
     else:
         names = list(Lx.keys())
+    for var in names:
+        Lx[var].encoding = {}
     Lx[names].to_netcdf(outfile, mode='w', format='NETCDF4', compute=True)
 
-def getColNames(vars_df, booms=None, data_type=None, bedrock=False):
+def getColNames(vars_df, ds, remove_nan_fields=False):
```

Comment:
It was meant to simplify the call to `getColNames` and make it easier to update without having to back-track all of its calls and update each of the call's inputs. Now we have different variables being written depending on whether it is a level 2/3, bedrock/mobile station, one/two booms... so it starts to be many inputs that need to be given to the function.

It makes it also more robust because, in its previous form and us, it would not work properly if the `boom` or `bedrock` attribute did not exist.

Reply:
See the comment above, I basically needed a function that could work on any given xr.Dataset object, even if it doesn't have `number_of_booms` or `data_type` attributes.
So now the existence of the attributes is tested within the function instead of before calling the function (simpler call to the function).

Reply:
Ah sorry for the bad explanations... There was obviously a difference between what I thought I was doing and what I ended up doing.

In the following
```
    vars_df = vars_df.loc[vars_df[ds.attrs['level']] == 1]

    # selecting variable list based on geometry
    if ds.attrs['level'] in ['L0', 'L1', 'L2']:
        if ds.attrs['number_of_booms']==1:
            vars_df = vars_df.loc[vars_df['station_type'].isin(['one-boom','all'])]
        elif ds.attrs['number_of_booms']==2:
            vars_df = vars_df.loc[vars_df['station_type'].isin(['two-boom','all'])]
```

I am assuming that all the dataset arriving at this point have a `level` attribute.

if that `level` is <=2 then the dataset should have a `station_type` attribute.

But again, I believe that this case handling is better suited here than before/outside the call to `getColNames`, which purpose is figuring out which columns should be written based on an input dataset.

I hope it makes sense.

Reply:
That  makes sense.

I'll suggest the following doctoring:

```python
    '''
    Get variable names for a given dataset with respect to its type and processing level
    
    The dataset must have the the following attributes: 
    * level
    * number_of_booms when the processing level is <= 2
    
    This is mainly for exporting purposes. 
   
   Parameters
   -----------
   .....
   '''
```